plugins/main/bridge/bridge.go

```diff
@@ -374,11 +376,45 @@ func setupVeth(netns ns.NetNS, br *netlink.Bridge, ifName string, mtu int, hairp
 		if err != nil {
 			return nil, nil, fmt.Errorf("failed to setup vlan tag on interface %q: %v", hostIface.Name, err)
 		}
+
+		if !preserveDefaultVlan {
```

Comment:
You could move this above the VLAN add and simply remove all VLAN configuration, without the condition of `info.Vid != uint16(vlanID)`